bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,782 @@
+            if (this.madokaSettings.getCommunicationControllerVersion() == null
+                    || this.madokaSettings.getRemoteControllerVersion() == null) {
+                submitCommand(new GetVersionCommand());
+            }
+            submitCommand(new GetIndoorOutoorTemperatures());
+            submitCommand(new GetOperationmodeCommand());
+            submitCommand(new GetPowerstateCommand()); // always keep the "GetPowerState" aftern the "GetOperationMode"
+            submitCommand(new GetSetpointCommand());
+            submitCommand(new GetFanspeedCommand());
+        }, 10, c.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("[{}] dispose()", super.thing.getUID().getId());
+
+        dispose(refreshJob);
+        dispose(commandExecutor);
+        dispose(currentCommand);
+
+        // Unsubscribe to characteristic notifications
+        if (this.device != null) {
+            BluetoothCharacteristic charNotif = this.device
+                    .getCharacteristic(DaikinMadokaBindingConstants.CHAR_NOTIF_UUID);
+
+            if (charNotif != null) {
+                @NonNull
+                BluetoothCharacteristic c = charNotif;
+                this.device.disableNotifications(c);
+            }
+        }
+
+        super.dispose();
+    }
+
+    private static void dispose(@Nullable ExecutorService executor) {
+        if (executor != null) {
+            executor.shutdownNow();
+        }
+    }
+
+    private static void dispose(@Nullable ScheduledFuture<?> future) {
+        if (future != null) {
+            future.cancel(true);
+        }
+    }
+
+    private static void dispose(@Nullable BRC1HCommand command) {
+        if (command != null) {
+            // even if it already completed it doesn't really matter.
+            // on the off chance that the commandExecutor is waiting on the command, we can wake it up and cause it to
+            // terminate
+            command.setState(BRC1HCommand.State.FAILED);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("[{}] Channel: {}, Command: {}", super.thing.getUID().getId(), channelUID, command);
+
+        if (command instanceof RefreshType) {
+            // The refresh commands are not supported in query mode.
+            // The binding will notify updates on channels
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case DaikinMadokaBindingConstants.CHANNEL_ID_SETPOINT:
+                try {
+                    QuantityType<?> setpoint = (QuantityType<?>) command;
+                    DecimalType dt = new DecimalType(setpoint.intValue());
+                    submitCommand(new SetSetpointCommand(dt, dt));
+                } catch (Exception e) {
+                    logger.info("Data received is not a valid temperature", e);
```

Comment:
btw, all those info loggings look a bit fishy to me. You are again doing a "catch all" here, but then claim that the issue is due to invalid data. If you are certain of that, you should imho be able to catch a more specific exception.

In case of an exception here: Can the situation be considered to be a bug that should be reported as an issue? I guess so. I would then even suggest to log it as a warning or even error, so that the user is clearly notified that something isn't working as expected and he needs to ask for help.

Reply:
I have updated with WARN level instead of debug. The point is - if the FW gets updated - I'm not sure the way values are encoded might change or not - so I prefer catching it.